tmt/steps/execute/__init__.py

```diff
@@ -326,6 +326,32 @@ def check_result_file(self, test: "tmt.Test") -> "tmt.Result":
             data['note'] = f"invalid test result '{result}' in result file"
         return tmt.Result(data, test=test)
 
+    def check_custom_results(self, test: "tmt.Test") -> List["tmt.Result"]:
+        """
+        Process custom results.yaml file created by the test itself.
+        """
+        self.debug("Processing custom results.yaml file created by the test itself.")
+
+        custom_results_path = os.path.join(
+            self.data_path(test, full=True),
+            tmt.steps.execute.TEST_DATA,
+            'results.yaml')
+
+        # Nothing to do if there's no result file
+        if not os.path.exists(custom_results_path):
+            raise tmt.utils.FileError
+
+        with open(custom_results_path) as custom_results_file:
+            results = tmt.utils.yaml_to_list(custom_results_file)
```

Comment:
Use 'safe' YAML typ. We can't trust user input.

Reply:
I edited `yaml_to_*` functions and put there implicit parameter `'safe'`